app/assets/javascripts/welcome.js.erb

```diff
@@ -1,121 +1,152 @@
-    infoWindow.open(map);
-  }
+  infoWindow.open(map);
+}
 
   // Adds a marker to the map.
-  function addMarker(location, map, topic) {
-    var marker = new google.maps.Marker({
-      position: location,
-      map: map,
-      icon: topic,
-      draggable:true
-    });
-    if (topic == '<%=image_url('0.png')%>'){
-      markers.push(marker);
-    }else{
-      savedmarkers.push(marker);
-    }
-    map.panTo(location);
+function addMarker(location, map, topic) {
+  var marker = new google.maps.Marker({
+    position: location,
+    map: map,
+    icon: topic,
+    draggable:true
+  });
+  if (topic == '<%=image_url('0.png')%>'){
+    markers.push(marker);
   }
+  map.panTo(location);
+}
 
-  function addCircles(radius,localtopic){
-    markers[0].setMap(null);
-    var center= markers[0].getPosition();
-    topic= localtopic;
-    addMarker(center, map,topic);
-
-    var cityCircle = new google.maps.Circle({
-      strokeColor: '#FDBD39',
-      strokeOpacity: 0.8,
-      strokeWeight: 2,
-      fillColor: '#FFCB03',
-      fillOpacity: 0.35,
-      map: map,
-      center: center,
-      radius: radius
-    });
+function addCircles(radius,localtopic, target_id) {
+  markers[0].setMap(null);
+  var center= markers[0].getPosition();
+  topic= localtopic;
+  //addMarker(center, map,topic);
+  var marker = new google.maps.Marker({
+    position: center,
+    map: map,
+    icon: topic,
+    draggable:true,
+    targetId: target_id
+  });
+  addCircleToaMarker(marker, radius, center,'#FDBD39','#FFCB03');
+  addMarkerListener(marker,radius,center);
+}
 
-  }
-  function load_user_targets() {
+function load_user_targets() {
   $.getJSON('/targets/list', function(targetsList) {
     jQuery.each(targetsList, function(index, target){
-
       var latitude= parseFloat(target.latitud);
       var longitude= parseFloat(target.longitud);
       var location = {lat: latitude, lng: longitude};
       var marker = new google.maps.Marker({
         position: location,
         map: map,
         icon: "/assets/"+target.topic_id+".png",
-        draggable:true
-      });
-      var cityCircle = new google.maps.Circle({
-        strokeColor: '#FDBD39',
-        strokeOpacity: 0.8,
-        strokeWeight: 2,
-        fillColor: '#FFCB03',
-        fillOpacity: 0.35,
-        map: map,
-        center: location,
-        radius: target.area
+        draggable:true,
+        targetId: target.id
       });
+      addCircleToaMarker(marker, target.area, location,'#FDBD39','#FFCB03');
+      addMarkerListener(marker,target.area,location);
     });
   });
 }
+
+function addMarkerListener(marker, area, location) {
+  marker.addListener('click', function() {
+    $.getJSON('/targets/load_delete_target', { target_id: marker.targetId} )
+      .done(function( data ) {
+        $(".rectangle").html(data.target);
+        marker.cityCircle.setMap(null);
+        var markertodelete= markerstodelete.pop();
```

Comment:
Use space when using =. ´var markertodelete = markerstodelete.pop(); ´ Check the javascript style guidelines. https://github.com/airbnb/javascript#whitespace--infix-ops